bpdv3/Assets/Scripts/enemy/Enemy.cs

```diff
@@ -38,8 +42,12 @@ public enum EnemyType
 
         private void OnDisable()
         {
-            token = true;
+            if (hasSpawned)
```

Comment:
this was the main reason for making the hasSpawned variable. Including this removes the error message that appeared in the console. You will see similar on the enemynextmove script also.